format.sh

```diff
@@ -38,11 +38,26 @@ source venv/bin/activate
 
 echo "Checking Python files..."
 if [ "$COMMAND" == "fix" ]; then
-  black .
+  black --quiet .
 else
   black --check --diff .
 fi
-pylint utils  # pylint does not have a fix mode
+
+# Create a temporary cache directory
+TEMP_CACHE_DIR=$(mktemp -d)
+
+if ! mypy --config pyproject.toml --install-types --non-interactive --cache-dir="$TEMP_CACHE_DIR"; then
+  echo "Mypy type checks failed. Please fix the errors above. 💥 💔 💥"
+  # Clean up the temporary cache directory
+  rm -rf "$TEMP_CACHE_DIR"
```

Comment:
Why is the purpose of this cache, if it's removed at the end of the process ?

Reply:
It was my attempt to fixing the linting problem because i thought it had to do with mypy cache -- with the temporary cache it clears all previous mypy cache issues

Reply:
I'm going to take it out since it's not solving the problem